library/oeUI/src/OemrUI.php

```diff
@@ -0,0 +1,290 @@
+    private $expand_icon_class;
+    private $expand_title;
+    private $expandable;
+    private $expandable_icon;
+    private $heading;
+    private $help_file;
+    private $help_icon;
+    private $help_modal;
+    private $jquery_draggable;
+    private $modal_body;
+    private $print;
+    private $target;
+       
+    /**
+    * Used to create the html string that will display the formatted heading with selected icons - expandable, action and help and generate the html code for the help modal.
+    *
+    * @param array $arrOeUiSettings is an associative array that contains 8 elements, string 'heading_title', int|bool 'expandable', int|bool 'current_state',
+    * string 'action', string 'action_title', string 'action_href', int|bool 'show_help_icon' and string 'help_file_name'.
+    * int|bool 'current_state' (expanded = 1, centered = 0)value is obtained from function collectAndOrganizeExpandSetting(array("")), this function needs an
+    * indexed array as an argument that contains the file name of the current file as the first element, the name of any other file that needs to open in a similar state
+    * needs to be included in this array,all names must be unique
+    * and have a '_xpd' suffix.
+    * It will be used to generate up to 4 values - string $heading, string $expandable_icon, string $action_icon and string $help_icon that will form the html string used to output
+    * the formatted heading of the page.
+    * If a feature is not required set the corresponding element in the array to an empty string
+    */
+    public function __construct($arrOeUiSettings = array())
+    {
+        $this->heading = $arrOeUiSettings['heading_title'];
+        $this->expandable = $arrOeUiSettings['expandable'];
+        $this->current_state = $arrOeUiSettings['current_state'];
+        $this->arrAction = array($arrOeUiSettings['action'], $arrOeUiSettings['action_title'], $arrOeUiSettings['action_href']);
+        $this->display_help_icon = $arrOeUiSettings['show_help_icon'];
+        $this->help_file = $arrOeUiSettings['help_file_name'];
+    }
+    
+    /**
+    * Used to create the html string that will display the formatted heading with selected icons - expandable, action and help.
+    *
+    * @return array containing string $heading - the formatted html string of the actual heading and string $container - the value of the container class 'container' or 'container-fluid'
+    *
+    */
+    public function pageHeading()
+    {
+        $heading = text($this->heading);
+        if (!empty($heading)) {
+            $arrexpandIcon = $this->expandIcon();// returns and array containing expandable icon string and container class string
+            $action_icon = $this->actionIcon();
+            $help_icon = $this->helpIcon();
+            $expandable_icon = $arrexpandIcon[0];
+            $container = $arrexpandIcon[1];
+            $heading = "<h2>$heading $expandable_icon $action_icon $help_icon</h2>";
```

Comment:
```
$heading = "<h2>" . text($heading) . " $expandable_icon $action_icon $help_icon</h2>";
```

Reply:
already done :)
$heading = text($this->heading);

Reply:
lol, missed that